cli/src/main/kotlin/commands/AnalyzerCommand.kt

```diff
@@ -148,5 +149,17 @@ class AnalyzerCommand : CliktCommand(name = "analyze", help = "Determine depende
             println("Writing analyzer result to '$file'.")
             file.mapper().writerWithDefaultPrettyPrinter().writeValue(file, ortResult)
         }
+
+        val analyzerResult = ortResult.analyzer?.result
+
+        if (analyzerResult == null) {
+            println("There was an error creating the analyzer result.")
+            throw ProgramResult(1)
+        }
+
+        if (analyzerResult.hasIssues) {
```

Comment:
It just occurred to me that it would actually be nicer to only return `2` here if there are *unresolved* issues. However, the analyzer does not know currently whether an issue is unresolved or not, as we're not passing a `ResolutionProvider`. While that's certainly technically possible to do, this is a bigger change that I want to discuss in a bigger round first.